### PR TITLE
Add mergePersistentFlags in strip flags since we now look at the flag set.

### PR DESCRIPTION
**Pull request by brendandburns**
@jlowdermilk @spf13 @smarterclayton ptal.

Thanks!
--brendan

### CODE REVIEW

**Comment by smarterclayton** [comment]
Test case? :)

**Comment by brendandburns** [comment]
Test cases added.

ptal.

Thanks!
--brendan

File: command_test.go [hunk 1]
```diff
@@ -68,6 +76,7 @@ func TestStripFlags(t *testing.T) {
 	var flagi int
 	var flagstr string
 	var flagbool bool
+	cmdPrint.PersistentFlags().BoolVarP(&flagbool, "persist", "p", false, "help for persistent one")
```
Comment (jlowdermilk) [inline command_test.go]: Can you add a subcommand of `cmdPrint` with a persistent flag, and verify that stripFlags works?

Comment (jlowdermilk) [inline command_test.go]: TODO is fine, thanks.

**Comment by jlowdermilk** [comment]
lgtm

**Pull request merged**